### PR TITLE
Mock & Live environment

### PR DESCRIPTION
**Pull request by gbasile**
### Description
Having explicit dependencies makes it possible to have more control over the environments that we want to use while developing. With this PR, it will be possible to change the dependencies for the Xcode previews so that it would be possible to work on the Gallery View without being logged in with a Google Account or relying on the data from Firebase. 

I can see this having a big impact on the 
- **Development speed** Because we can simply use a mock environment in order to easily create the conditions that we want to recreate on the environment. (I can simply test Step 5 of a long onboarding process without having to start from 1)
- **Compilation time** All the mocks environments should not rely on the live dependencies, in our case we can test the Gallery without having to compile the Firebase and Google SDK
- **Design Friendly** With the power of Xcode 12 realtime preview and because Swift UI semantic language it should be easy for a designer to jump into a specific section of the app and test different UI styles without having to lunch the app or deploy the app into a simulator/real device

### What's Next
- Modularisation: so that we can run the mocks environments without any external dependency
- Static Image support for the mocked users. In this way, we could also consider networking as an external dependency 
- Tuist? To make modularisation easy peasy 

### I want to know more

I highly recommend you to watch this collection of videos about dependencies from [Pointfree](https://www.pointfree.co/collections/dependencies)

This is the app running against mocked data:

### CODE REVIEW

File: ios/Peepz/Peepz/PeepzApp.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ class AppDelegate: NSObject, UIApplicationDelegate {
 @main
 struct PeepzApp: App {
     @UIApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
-    @StateObject private var model = PeepzModel()
+    @StateObject private var model: PeepzModel = .live
```
Comment (gbasile) [inline ios/Peepz/Peepz/PeepzApp.swift:21]: Here we can simply change to `.mock` to use static data instead and skip the Google Login

Comment (wrumble) [reply ios/Peepz/Peepz/PeepzApp.swift:21]: could we trigger this from app settings when in dev? that would be a cool addition

**Review by wrumble — APPROVED** [review]
Looks amazing! Tuist scares me, mostly because of how much time I and others lost due to carthage issues 🙈 on the Crvsh project.

File: ios/Peepz/Peepz/PeepzApp.swift (hunk 1 above)
Comment (gbasile) [reply ios/Peepz/Peepz/PeepzApp.swift:21]: yes, I think it's definitely something that would be worth to investigate 👍

**Review by Michal-Novoda — APPROVED** [review]

**Pull request merged**